application/src/main/java/org/togetherjava/tjbot/commands/tags/TagCommand.java

```diff
@@ -46,20 +50,40 @@ public void onSlashCommand(SlashCommandInteractionEvent event) {
         String id = Objects.requireNonNull(event.getOption(ID_OPTION)).getAsString();
         OptionMapping replyToUserOption = event.getOption(REPLY_TO_USER_OPTION);
 
-        if (tagSystem.handleIsUnknownTag(id, event)) {
+        if (tagSystem.handleIsUnknownTag(id, event, super::generateComponentId,
+                replyToUserOption)) {
             return;
         }
 
-        ReplyCallbackAction message = event
-            .replyEmbeds(new EmbedBuilder().setDescription(tagSystem.getTag(id).orElseThrow())
-                .setFooter(event.getUser().getName() + " • used " + event.getCommandString())
-                .setTimestamp(Instant.now())
-                .setColor(TagSystem.AMBIENT_COLOR)
-                .build());
+        sendTagReply(event, event.getUser().getName(), id, Optional.of(event.getCommandString()),
+                Optional.ofNullable(replyToUserOption)
+                    .map(OptionMapping::getAsUser)
+                    .map(IMentionable::getAsMention));
+    }
 
-        if (replyToUserOption != null) {
-            message = message.setContent(replyToUserOption.getAsUser().getAsMention());
+    @Override
+    public void onButtonClick(ButtonInteractionEvent event, List<String> args) {
+        if (!TagSystem.TAG_SUGGESTION_INDICATOR.equals(args.get(0))) {
```

Comment:
thats really dirty. also, nothing in the code prevents someone from creating a tag with that name and then everything goes bonkers.

Reply:
how would I do it instead then?

Reply:
maybe i just misunderstood what this does. why do u need it in the first place? this argument seems completely obsolete to me

Reply:
to know if the button comes from a suggestion. otherwise it'd throw lots of errors if a button was clicked that doesn't come from us because the arguments are wrong. or am I misunderstanding when onButtonClick gets invoked?

Reply:
> or am I misunderstanding when onButtonClick gets invoked?

yes. its only invoked for buttons created by ur class. ur safety check is obsolete. `BotCore` already takes care of this for u.

Reply:
done